pygridgen/tests/test_focus.py

```diff
@@ -156,3 +156,25 @@ def test_focuspoint_to_dict():
     fp2 = pygridgen.grid._FocusPoint(**fp.to_dict())
     assert fp.to_dict() == {'pos': 0.25, 'axis': 'x', 'factor': 4, 'extent': 0.5}
     assert fp.to_dict() == fp2.to_dict()
+
+
+def test_focus_to_from_spec():
+    fpx = pygridgen.grid._FocusPoint(0.25, 'x', 4, 0.5)
+    fpy = pygridgen.grid._FocusPoint(0.33, 'y', 0.1, 0.2)
+    f1 = pygridgen.grid.Focus(fpx, fpy)
+
+    # Check that the spec dictionary returns desired results
+    assert f1.to_spec() == [
+        {'pos': 0.25, 'axis': 'x', 'factor': 4, 'extent': 0.5},
+        {'pos': 0.33, 'axis': 'y', 'factor': 0.1, 'extent': 0.2}
+    ]
+
+    f2 = pygridgen.grid.Focus.from_spec(f1.to_spec())
+
+    # Check that the spec dictionary returns desired results of the orignal
+    # focus and the focus generated from .to_spec()
+    # First check the keys are in order and then check the values
```

Comment:
order of the keys doesn't matter. `assert f2.to_spec() == f1.to_spec()` should suffice there.